src/django_pg_migration_tools/operations.py

```diff
@@ -154,16 +116,116 @@ def _ensure_original_lock_timeout_is_reset(
             self.SET_LOCK_TIMEOUT_QUERY, {"lock_timeout": self.original_lock_timeout}
         )
 
-    # The following methods are necessary for Django to understand state
-    # changes.
-    def state_forwards(
-        self, app_label: str, state: migrations.state.ProjectState
+
+class SaferAddIndexConcurrently(
+    BaseIndexOperation, psql_operations.AddIndexConcurrently
```

Comment:
is there anything stopping us from not using inheritence for composition? So that the SaferAddIndexConcurrently instance creates it's own instance of AddIndexConcurrently that it delegates to rather than being a subclass of AddIndexConcurrently?

Reply:
I'd surely prefer composition if the code action was "AddIndexConcurrently, please go do something for me".

But in this case the code action is more like "AddIndexConcurrently, I need your interface so that Django's migration apparatus can understand my as an index creation operation".

This includes having methods that will register application database state alterations for example. But more importantly, I'm interested in picking up changes from upstream on that specific interface.

For example, a recent change in Django (new in 5.1) has added the "category" attribute to operations. In this case, `category = OperationCategory.ADDITION` (or REMOVAL), wouldn't be picked up by our classes unless we inherit from upstream. Otherwise, we'd have to keep different interfaces for different Django versions, which is less desirable.

Reply:
fair enough, inheritence sounds reasonable then.